docs/source/guide/s3-example-creating-buckets.rst

```diff
@@ -33,20 +34,30 @@ platform.
     from botocore.exceptions import ClientError
 
 
-    def create_bucket(bucket_name):
-    """ Create an Amazon S3 bucket
-
-    :param bucket_name: Unique string name
-    :return: True if bucket is created, else False
-    """
-
-    s3 = boto3.client('s3')
-    try
-        s3.create_bucket(Bucket=bucket_name)
-    except ClientError as e:
-        logging.error(e)
-        return False
-    return True
+    def create_bucket(bucket_name, region=None):
+        """Create an S3 bucket in a specified region
+
+        If a region is not specified, the bucket is created in the S3 default
+        region (us-east-1).
+
+        :param bucket_name: Bucket to create
+        :param region: String region to create bucket in, e.g., 'us-west-2'
+        :return: True if bucket created, else False
+        """
+
+        # Create bucket
+        s3_client = boto3.client('s3')
```

Comment:
We should also set the client `region_name` when instantiating the client. While the client region does not necessarily need to be set for the main aws partition, the region needs to be set if the region is not in the main aws partition (e.g. `cn-north-1`) as the client will need to direct the request to the proper endpoint.

Reply:
I updated the code sample to set the S3Client `region_name` argument.